test-basic/unnestTest.js

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2021 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const assert = require('assert');
+const marklogic = require('../');
+const testconfig = require('../etc/test-config.js');
+const db = marklogic.createDatabaseClient(testconfig.restWriterConnection);
+const op = marklogic.planBuilder;
+
+const TEAM_MEMBER_NAME_COLUMN = "teamMemberName";
+
+
+describe('test unnest function', function () {
+
+    before(function (done) {
+        const doc = {
+            office: [
+                { department: "Engineering", teamMembers: "Cindy,Alice,Dan" },
+                { department: "Sales", teamMembers: "Bob" },
+                { department: "Marketing", teamMembers: null }
+            ]
+        };
+
+        db.documents.write({
+            uri: '/optic/test/office1.json',
+            contentType: 'application/json',
+            content: doc
+        }).result(function() {
+            done();
```

Comment:
Just curious, is it necessary to call `done()` here? (I'm not familiar with tests in this project, so I'm asking so I know if this needs to be done when I try to write a test)

Reply:
Hi Rob, yes done() is needed towards the end else the test hangs saying - "Timeout  exceeded. For async tests and hooks, ensure "done()" is called;". done() is needed individually for all the blocks including before, beforeEach, after, afterEach and the tests.